### DISCUSSION TITLE
Using ESP8266 sockets wit HomeSpan

### DISCUSSION THREAD

**Discussion by homonto** [HomeSpan/HomeSpan#709] [General]
hi @HomeSpan 
8266 eventually not compatible, right?
many sockets are esp 8266 based - it would be nice to bring them to HomeKit using your library  not ESPNow, rather always ON because then need to react to ON/OFF)

**Comment by HomeSpan** [comment]
sorry, but no plans to port HomeSpan to 8266

**Reply by homonto** [reply]
oki doki, thank you

**Reply by YouCanNotBeSerious** [reply]
I spent time researching the 8266 situation, and it seems like there are a lot of tunings and compromises to be made to have HomeKit work on esp8266 due to the constraints.  Lots or memory and heap things, and cryptography and stuff way over my current understanding and ability!

IMHO better to just use esp32 where you can, but I was also in the position of wanting to use sonoff 8266 (actually 8285) based switches, and I found this if you want to use Arduino: https://github.com/Mixiaoxiao/Arduino-HomeKit-ESP8266
or this if you want a GUI to configure the switch:
https://github.com/RavenSystem/esp-homekit-devices

Of those, I prefer the second one - HAA for use on esp8266. Works great on sonoff basic type switches and should on many others.  Even then it was crashing for me until I turned off network logging on the advice of the dev - again, the 8266 is limited.

Of ALL the options given free choice, I have found HomeSpan on esp32 to be the best overall so for any devices I build myself, I use it.
